olive/systems/azureml/aml_system.py

```diff
@@ -621,5 +644,71 @@ def _create_metric_component(
 
         return metric_component
 
+    def get_supported_execution_providers(self) -> List[str]:
```

Comment:
I'm not sure we should create a new job just for available ep fetching. I prefer 2 options here: 1. user still needs to provide desired ep for aml system. 2. somehow update our aml runner script logic to support testing all available eps together in one script.

Reply:
Yeah, I agree with xiaoyu. The dedicated job seems a little bit heavy. for azureml/docker system, I still prefer let user specify the EP explicitly to simplify the get available EP logic.

Reply:
I create PR https://github.com/microsoft/Olive/pull/592 for same purpose.

Reply:
Thanks this makes sense to me. I rebased to main with your merged PR. 
I think we can still merge this PR since I made some code improvements to the systems and the `get_supported_execution_providers` methods is still useful. In case the user wants to debug and check the available eps in a remote system.

Reply:
I am not still sure whether we need get available EPs for both AzureML and docker. 
better confirm with @devang-ml

Reply:
I think it's okay to have this method. Might be useful in the future or if a user wants to check what eps their remote system supports.

Reply:
It's already implemented and doesn't change any of the current engine or olive behavior.